pkg/beholder/example_test.go

```diff
@@ -0,0 +1,89 @@
+package beholder_test
+
+import (
+	"context"
+	"log"
+	"math/rand"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/protobuf/proto"
+
+	// chainlink-common
+	beholder "github.com/smartcontractkit/chainlink-common/pkg/beholder/global"
+	"github.com/smartcontractkit/chainlink-common/pkg/beholder/pb"
+)
+
+func ExampleBeholderCustomMessage() {
+	beholderConfig := beholder.NewConfig()
+
+	// Bootstrap Beholder Client
+	err := beholder.Bootstrap(beholderConfig, errorHandler)
+	if err != nil {
+		log.Fatalf("Error bootstrapping Beholder: %v", err)
+	}
+
+	// Define a custom protobuf payload to emit
+	payload := &pb.TestCustomMessage{
+		BoolVal:   true,
+		IntVal:    42,
+		FloatVal:  3.14,
+		StringVal: "Hello, World!",
+	}
+	payloadBytes, err := proto.Marshal(payload)
+	if err != nil {
+		log.Fatalf("Failed to marshal protobuf")
+	}
+
+	// Emit the custom message anywhere from application logic
+	for range 10 {
+		err := beholder.Emit(context.Background(), payloadBytes,
+			"beholder_data_schema", "/custom-message/versions/1", // required
+			"beholder_data_type", "custom_message",
```

Comment:
Is `beholder_data_type` just an example (i.e. it's just as meaningful as `foo bar`)?

Reply:
Yes its just an example

Reply:
yes, its just an example